### PR TITLE
py-sphinx: update to 4.3.0 and 3.5.4

### PR DESCRIPTION
**Pull request by catap**
#### Description

py-sphinx: update to 4.3.0

py35-sphinx: update to 3.5.4, also backported support of docutils-0.18

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 11.6 20G165 x86_64
Xcode 13.1 13A1030d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by jmroot** [comment]
Thanks. Looks good as long as all the upstream docutils fixes were released in 4.3.0? However py310-sphinx is failing; looks like there are some dependencies that don't have a py310 subport yet.

**Comment by jmroot** [comment]
```
% port lint subport=py310-sphinx
--->  Verifying Portfile for py310-sphinx
Error: Unknown dependency: py310-babel
Error: Unknown dependency: py310-sphinxcontrib-applehelp
Error: Unknown dependency: py310-sphinxcontrib-devhelp
Error: Unknown dependency: py310-sphinxcontrib-htmlhelp
Error: Unknown dependency: py310-sphinxcontrib-jsmath
Error: Unknown dependency: py310-sphinxcontrib-serializinghtml
Error: Unknown dependency: py310-sphinxcontrib-qthelp
--->  7 errors and 0 warnings found.
```

**Comment by catap** [comment]
@jmroot I forgot to add to PR all related commits. Let me fix it.

Yes, all fixes for support of docutils-0.18 was merged 10 days ago and yesterday a new release is happened. All these changes are part of this release.

Anyway, I made small backporting to the patch to version 3.5.4 which is used for python 3.5: https://github.com/catap/sphinx/tree/docutils-1.8-v354

**Review by jmroot — APPROVED** [review]
Thanks. Looks fine to merge if CI passes and @stromnov has no objections.

**Comment by catap** [comment]
CI is failed on py-jinja2, let me rebase to last master

File: python/py-sphinxcontrib-qthelp/Portfile [hunk 1]
```diff
@@ -23,7 +23,7 @@ checksums           rmd160  b41a6b415c5653b0828009bcd73b18823d3cc1b8 \
                     sha256  4c33767ee058b70dba89a6fc5c1892c0d57a54be67ddd3e7875a18d14cba5a72 \
                     size    21658
 
-python.versions     35 36 37 38 39
+python.versions     35 36 37 38 39 310
```
Comment (reneeotten) [inline python/py-sphinxcontrib-qthelp/Portfile:26]: this has the wrong commit message

**Comment by reneeotten** [comment]
typically people would have a commit message like for example: `py-latexcodec: add py310 subport`; but I guess the way you have them would work as well

**Comment by catap** [comment]
@reneeotten commit message  are rewritten.

**Pull request merged**